src/V8JsiRuntime.cpp

```diff
@@ -606,24 +599,42 @@ jsi::Value V8Runtime::evaluateJavaScript(
 
   _ISOLATE_CONTEXT_ENTER
 
-  v8::Local<v8::String> sourceV8String;
-  // If we'd somehow know the buffer includes only ASCII characters, we could use External strings to avoid the copy
-  /* ExternalOwningOneByteStringResource *external_string_resource = new ExternalOwningOneByteStringResource(buffer);
-  if (!v8::String::NewExternalOneByte(isolate, external_string_resource).ToLocal(&sourceV8String)) {
-    std::abort();
+  // Basic hashing of the buffer for cache versioning
+  constexpr std::uint64_t P1 {7};
+  constexpr std::uint64_t P2 {31};
+
+  std::uint64_t hash {P1};
+  bool isAscii {true};
+
+  const char* p = reinterpret_cast<const char *>(buffer->data());
+  for (size_t i=0; i < buffer->size(); p++, i++) {
+    hash = hash * P2 + *p;

```

Comment:
Tudor, Is this a well-know algorithm ? If i understand correctly, it loops through each chars multiplying the accumulator with 31 (left shift 5 times and subtract once ?) and add current char to accumulator .. I see it might work as the result might depend on all the previous chars .. One possible drawback would be that there is 'n' multiplications here .. 

Did you consider alternatives, such as this one: [https://en.wikipedia.org/wiki/Jenkins_hash_function#one_at_a_time](https://en.wikipedia.org/wiki/Jenkins_hash_function#one_at_a_time)

I think, shifting, xoring and adding is a lot more performant than multiplications ..

Reply:
No, this is just my go-to default hashing algorithm with primes (non-crypto) when I just need uniqueness. I'm open to whatever algorithm makes sense as long as it's small and unobtrusive (we don't need crypto / signing, just uniqueness).

Reply:
FWIW, just benchmarked the silly implementation against the Jenkins, and Jenkins is 2.5x slower.

Reply:
I found hashing algorithms that are even faster (Murmur, Meow, Spooky) but the source code is huge.

Reply:
This looks like a black hole of research, I think I'll just use the std::hash<> implementation and don't bother optimizing.

Reply:
May I suggest using the std::hash algorithm instead - it is basically the FNV-1 algorithm which is known to be quite fast and has a good distribution.